tools/depends/native/TexturePacker/src/decoder/GifHelper.cpp

```diff
@@ -18,10 +18,10 @@
  *
  */
 
-#include "GifHelper.h"
 #include <algorithm>
 #include <stdlib.h>
 #include <cstring>
+#include "GifHelper.h"
```

Comment:
If you have to have the header at a certain place to get stuff compiled, this might be an indicator that the root cause of the problem lies in GifHelper.h or in one of the headers included by GifHelper.h. This fix looks a little bit hacky to me, tbh.

Reply:
Is this actually needed?

Reply:
The root cause is that reallocarray is a new function in glibc 2.26 which is clashing with https://sourceforge.net/p/giflib/code/ci/master/tree/lib/gif_lib.h#l248

Reply:
> This fix looks a little bit hacky to me, tbh.

I don't disagree, but I'm not a C++ coder so I'm not entirely sure what would be the best fix. This seemed like a quick fix without going too far down the rabbit hole. :)

Reply:
giflib needs to be fixed to use system (glibc) reallocarray if available or at least to change signature of it's own reallocarray function to not clash with system (glibc) reallocarray declaration.

Reply:
while you are correct in that being the correct fix, sometimes some pragmatism must be applied. this fixes it in a non-intrusive way for the time being, on all platforms. this is gcc vs giflib issue and it will bite on native builds on 17.10 as well..

Reply:
Sure @notspiff, I'm only commenting here to make sure we understand the actual problem. Im in no way blocking this PR.